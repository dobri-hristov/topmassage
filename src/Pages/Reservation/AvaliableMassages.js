import React, { useState } from 'react'
import InputLabel from '@mui/material/InputLabel';
import FormControl from '@mui/material/FormControl';
import Select from '@mui/material/Select';
import MenuItem from '@mui/material/MenuItem';

const AvaliableMassages = ({ allMassages, isAuth}) => {
    const [massage, setMassage] = useState('');

    const handleMassage = (event) => {
        setMassage(event.target.value)
    }

    return (
        <FormControl variant="standard" fullWidth>
            <InputLabel>Вид масаж</InputLabel>
            <Select
                value={massage}
                name="massage"
                onChange={handleMassage}
                fullWidth
                id="reservationMassage"
            >
                {
                    allMassages.map((massage) => (
                        Object.keys(massage.prices).map((minutes, index) => {
                            const price = Number(massage.prices[minutes])
                            const discountPrice = price - ((price / 100) * 20)

                            if (isAuth) {
                                return <MenuItem key={index} value={`${massage.massageName}-${minutes}мин.-${discountPrice}лв.`}>
                                    {`${massage.massageName} - ${minutes} мин./`}
                                    <span style={{ "color": "red" }}>{discountPrice} лв.</span>
                                </MenuItem>
                            }

                            return <MenuItem key={index} value={`${massage.massageName}-${minutes}мин.-${price}лв.`}>
                                {`${massage.massageName} - ${minutes} мин. - ${price} лв.`}
                            </MenuItem>
                        })
                    ))
                }
            </Select>
        </FormControl>
    )
}

export default AvaliableMassages
