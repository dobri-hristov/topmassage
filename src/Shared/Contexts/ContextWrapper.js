import React, { createContext, useState, useEffect } from 'react'
import { getAuth, onAuthStateChanged } from "firebase/auth";

export const AuthContext = createContext({
    auth: '',
    username: '',
    isLoaded: false,
})

const ContextWrapper = (props) => {
    const [auth, setAuthUser] = useState('')
    const [username, setUsername] = useState('')
    const [isLoaded, setIsLoaded] = useState(false)
    const authUser = getAuth()

    useEffect(() => {
        onAuthStateChanged(authUser, (user) => {
            if (user != null) {
                setUsername('')
                const injectUsername = () => {
                    if (user.displayName === null) {
                        setTimeout(() => {
                            injectUsername()
                        }, 500)
                    } else {
                        setUsername(user.displayName)
                        setIsLoaded(true)
                    }
                }
                injectUsername()
                setAuthUser(true)
            } else {
                setAuthUser(false)
                setIsLoaded(true)
            }
        })
    }, [authUser])

    return (
        <AuthContext.Provider value={{ auth, username, isLoaded }}>
            {props.children}
        </AuthContext.Provider>
    )
}

export default ContextWrapper
