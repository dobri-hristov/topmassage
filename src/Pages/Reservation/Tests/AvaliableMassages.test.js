import React from 'react'
import TestRenderer from 'react-test-renderer'
import AvaliableMassages from '../AvaliableMassages.js'
import { AuthContext } from "../../../Shared/Contexts/ContextWrapper";


describe('Reservation page/AvaliableMassages component', () => {
    test('should render correctly with no data', () => {
        const component = TestRenderer.create(<AvaliableMassages allMassages={[]} />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    test('should render correctly with data and user is auth', () => {
        const data = [
            {
                massageName: 'Класически масаж',
                prices: {
                    60: 90,
                    30: 60
                }
            }
        ]
        const component = TestRenderer.create(
            <AuthContext.Provider value={{ auth: true }}>
                <AvaliableMassages allMassages={data} />
            </AuthContext.Provider>
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    test('should render correctly with data and user is NOT auth', () => {
        const data = [
            {
                massageName: 'Класически масаж',
                prices: {
                    60: 90,
                    30: 60
                }
            }
        ]
        const component = TestRenderer.create(
            <AvaliableMassages allMassages={data} />
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})