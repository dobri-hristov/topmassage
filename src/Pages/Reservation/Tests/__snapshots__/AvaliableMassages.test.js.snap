// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservation page/AvaliableMassages component should render correctly with data and user is NOT auth 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
  >
    Вид масаж
  </label>
  <div
    className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiSelect-root css-ghsjzk-MuiInputBase-root-MuiInput-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="reservationMassage"
      className="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
      id="reservationMassage"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="notranslate"
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      name="massage"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value=""
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`Reservation page/AvaliableMassages component should render correctly with data and user is auth 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
  >
    Вид масаж
  </label>
  <div
    className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiSelect-root css-ghsjzk-MuiInputBase-root-MuiInput-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="reservationMassage"
      className="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
      id="reservationMassage"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="notranslate"
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      name="massage"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value=""
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`Reservation page/AvaliableMassages component should render correctly with no data 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
  >
    Вид масаж
  </label>
  <div
    className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiSelect-root css-ghsjzk-MuiInputBase-root-MuiInput-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="reservationMassage"
      className="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
      id="reservationMassage"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="notranslate"
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      name="massage"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value=""
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
